src/lib/components/buttons/Button.svelte

```diff
@@ -0,0 +1,104 @@
+<script lang="ts">
+  import Spinner from "../Transition/Spinner.svelte";
+
+  export let title = "Submit";
+  export let onClick: (e) => void;
+  export let loaderSize = 16;
+  export let disable = false;
+  export let loader = false;
+  export let buttonStyleProp = "";
+  export let buttonClassProp = "";
+  export let textClassProp = "";
+  export let textStyleProp = "";
+  export let allowChild = false;
+  export let type:
+    | "primary"
+    | "dark"
+    | "danger"
+    | "transparent"
+    | "other"
+    | "icon" = "other";
+  enum BtnType {
+    PRIMARY = "primary",
+    DARK = "dark",
+    DANGER = "danger",
+    TRANSPARENT = "transparent",
+    OTHER = "other",
+    ICON = "icon",
+  }
+  let btnClass = "";
+  if (type === BtnType.PRIMARY) {
```

Comment:
Can we use switch here for better readability. Also put it in a different function. We should also handle default condition when no type is provided.

switch (type) {
  case BtnType.PRIMARY:
    btnClass = "custom-btn-primary";
    break;
  case BtnType.DARK:
    btnClass = "custom-btn-dark";
    break;
  case BtnType.DANGER:
    btnClass = "custom-btn-danger";
    break;
  case BtnType.TRANSPARENT:
    btnClass = "custom-btn-transparent";
    break;
  case BtnType.ICON:
    btnClass = "sparrow-icon-btn";
    break;
  default:
    // handle default case if necessary
    break;
}

Reply:
Done, for default passing empty in case of no class.